target/linux/airoha/files/drivers/pinctrl/pinctrl-airoha.c

```diff
@@ -2189,6 +2353,63 @@ static int airoha_pinctrl_probe(struct platform_device *pdev)
 	if (err)
 		return err;
 
+	spin_lock_init(&pinctrl->gpiochip.lock);
+
+	if (of_find_property(np, "interrupt-controller", NULL)) {
+		struct gpio_irq_chip *girq = NULL;
+		int ret = platform_get_irq(pdev, 0);
+
+		if (ret < 0) {
+			dev_err(dev, "Error requesting IRQ %d\n", ret);
+			return ret;
+		}
+		pinctrl->gpiochip.gpio_irq = ret;
+
+		pinctrl->gpiochip.irqstatus = devm_platform_ioremap_resource(pdev, index++);
+		if (IS_ERR(pinctrl->gpiochip.irqstatus))
+			return PTR_ERR(pinctrl->gpiochip.irqstatus);
+
+		pinctrl->gpiochip.level[0] = devm_platform_ioremap_resource(pdev, index++);
+		if (IS_ERR(pinctrl->gpiochip.level[0]))
+			return PTR_ERR(pinctrl->gpiochip.level[0]);
+
+		pinctrl->gpiochip.level[1] = devm_platform_ioremap_resource(pdev, index++);
+		if (IS_ERR(pinctrl->gpiochip.level[1]))
+			return PTR_ERR(pinctrl->gpiochip.level[1]);
+
+		pinctrl->gpiochip.edge[0] = devm_platform_ioremap_resource(pdev, index++);
+		if (IS_ERR(pinctrl->gpiochip.edge[0]))
+			return PTR_ERR(pinctrl->gpiochip.edge[0]);
+
+		pinctrl->gpiochip.edge[1] = devm_platform_ioremap_resource(pdev, index++);
+		if (IS_ERR(pinctrl->gpiochip.edge[1]))
+			return PTR_ERR(pinctrl->gpiochip.edge[1]);
+
```

Comment:
@merbanan I think we can have a better code architecture here. Let's add IRQ support for bank 1 and then we will evaluate

Reply:
Ack.